### PR TITLE
fix cmake issue with O2_GENERATE_LIBRARY

### PR DESCRIPTION
**Pull request by sawenzel**
* do no longer reset variables gives to O2_GENERATE_LIBRARY
 * resulting simplification of CMakeLists.txt files

see issue #149

### CODE REVIEW

**Comment by sawenzel** [comment]
@Barthelemy: Is this ok?

File: cmake/O2Utils.cmake [hunk 1]
```diff
@@ -280,14 +280,6 @@ macro(O2_GENERATE_LIBRARY)
   # Install all the public headers
   install(DIRECTORY include/${MODULE_NAME} DESTINATION include)
 
-  Set(LIBRARY_NAME)
-  Set(DICTIONARY)
```
Comment (matthiasrichter) [inline cmake/O2Utils.cmake]: It does not make sense to delete the library name at the end of the macro, cleaning all the other variables might be useful however.

**Comment by Barthelemy** [comment]
@matthiasrichter It was a copy paste from FairRoot. No reason that I am aware of. I would be in favour of not touching input variables at all, i.e. not deleting them, because it is unexpected for the caller. At least the library name should not be wiped out.

**Comment by Barthelemy** [comment]
@sawenzel all good for me :+1:

**Comment by Barthelemy** [comment]
I am ok with merging. I don't have the access rights to do it.

**Pull request merged**